test/document_stores/test_milvus.py

```diff
@@ -0,0 +1,22 @@
+from unittest import mock
+
+import pytest
+
+from packaging import version
+
+import haystack
+from haystack.document_stores.milvus import Milvus2DocumentStore
+
+fail_in_v1_13 = pytest.mark.xfail(
+    version.parse(haystack.__version__) > version.parse("1.12"),
+    reason="feature should be removed in v1.13, as it was deprecated in v1.11",
+)
+
+
+@fail_in_v1_13
+@mock.patch("haystack.document_stores.milvus.connections")
+@mock.patch("haystack.document_stores.milvus.MilvusDocumentStore._create_collection_and_index")
+def test_Milvus2DocumentStore_deprecated(_create_collection_and_index, connections):
+    print(version.parse(haystack.__version__), version.parse("1.10"))
+    with pytest.warns(FutureWarning):
+        Milvus2DocumentStore()
```

Comment:
Ok I see what is going on here, but in general I think this can't be handled this way.

The fact is that Milvus1 and 2 users were using the same exact code. Both were doing `from haystack.document_stores import MilvusDocumentStore`. So now adding a deprecation warning for users of the Milvus2 version is going to be extremely confusing, as we're in fact removing Milvus1 but we have no way to tell who was using what.

I propose for this specific docstore to ignore the usual deprecation policy and simply drop Milvus 1. The usage data shows close to no usage anyway. Doing it right would be a pain and I think it's just not worth the effort.

Reply:
The deprecation notice would only be visible to those importing Milvus2 explicitly:
```from haystack.document_stores.milvus import Milvus2DocumentStore```
which is not supposed to happen anyways.

If we drop this whole thing, I would also remove `Milvus2DocumentStore` altogether, I'm fine with both.

Reply:
Yes I'd definitely remove that shell class as well